src/contracts/core/AllocationManager.sol

```diff
@@ -53,13 +53,18 @@ contract AllocationManager is
     function slashOperator(
         SlashingParams calldata params
     ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
+        // Assert that the amount to slash is within bounds of 1 and WAD (1e18).
         require(0 < params.wadToSlash && params.wadToSlash <= WAD, InvalidWadToSlash());
 
-        // Check that the operator is registered and slashable
         OperatorSet memory operatorSet = OperatorSet({avs: msg.sender, operatorSetId: params.operatorSetId});
-        bytes32 operatorSetKey = _encodeOperatorSet(operatorSet);
+
+        // Assert that the provided strategies are all whitelisted for the given operator set.
+        require(avsDirectory.isOperatorSetStrategyBatch(operatorSet, params.strategies), InvalidStrategy());
```

Comment:
Can we not have this method accept an operator set as an argument and fetch strategies from the AVSD? That seems far preferable to passing in an array of strategies here.

Reply:
FIxed in #855, but the fix introduces a via-ir compile error. Thinking we wait until #845 is merged which should reduce code complexity.